localstack/services/sqs/provider.py

```diff
@@ -121,6 +122,71 @@ def check_message_content(message_body: str):
         raise InvalidMessageContents(error)
 
 
+class CloudwatchPublishWorker:
+    """
+    Regularly publish metrics data about approximate messages to Cloudwatch.
+    Includes: ApproximateNumberOfMessagesVisible, ApproximateNumberOfMessagesNotVisible
+        and ApproximateNumberOfMessagesDelayed
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.scheduler = Scheduler()
+        self.thread: Optional[FuncThread] = None
+
+    def publish_approximate_cloudwatch_metrics(self):
+        for account_id, region_bundle in sqs_stores.items():
+            for region, store in region_bundle.items():
+                for queue in store.queues.values():
+                    self.publish_approximate_metrics_for_queue_to_cloudwatch(queue)
+
+    def publish_approximate_metrics_for_queue_to_cloudwatch(self, queue):
+        """Publishes the metrics for ApproximateNumberOfMessagesVisible, ApproximateNumberOfMessagesNotVisible
+        and ApproximateNumberOfMessagesDelayed to CloudWatch"""
+        # TODO ApproximateAgeOfOldestMessage is missing
+        #  https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-available-cloudwatch-metrics.html
+        publish_sqs_metric(
+            region=queue.region,
+            queue_name=queue.name,
+            metric="ApproximateNumberOfMessagesVisible",
+            value=queue.visible._qsize(),
```

Comment:
could we add these metrics as attributes to the queue? then we have them in one place! i think we need them in some SQS api calls as well, when we return the queue attributes.